operations/update_by_id.ts

```diff
@@ -206,7 +309,7 @@ export const mapAccessPatterns = (
         'partition',
         partitionKeys,
         layout,
-        updates,
+        changes,
         collection.layout.indexKeySeparator
       );
       if (update) {
```

Comment:
When a sparse partition key value is deleted, shouldn't this delete the value as well?  Might be worth writing a unit test that covers update_by_id with the expected output for deleting a sparse partition/sort key.